src/udp_capture.cpp

```diff
@@ -595,6 +605,56 @@ class BFudpcapture_impl {
 		size_t total_span   = contig_span * 4;
 		size_t nringlet_max = 1;
 		_ring.resize(contig_span, total_span, nringlet_max);
+		this->open_logs();
+	}
+	inline void open_logs() {
+		// Get the PID
+		this->_pid = getpid();
+		
+		// Make the directory for this process
+		std::string log_dir = "/dev/shm/bifrost/"+std::to_string(this->_pid);
+		int status = system(("mkdir -p "+log_dir).c_str());
+		if( status != 0 ) {
+			// TODO: What to do here?
+			throw std::runtime_error("Cannot create capture status directory");
+		}
+		
+		// Open the log files
+		//// Capture type
+		_type_log.open((log_dir+"/type").c_str(), std::fstream::out);
+		///// Size information
+		_size_log.open((log_dir+"/sizes").c_str(), std::fstream::out);
+		//// Channel information
+		_chan_log.open((log_dir+"/chans").c_str(), std::fstream::out);
+		//// Capture statistics
+		_stat_log.open((log_dir+"/stats").c_str(), std::fstream::out);
+		
+		// Fill in known parameters which don't change
+		_type_log.seekp(0);
+		_type_log << "TYPE = "
+			     << "chips" << endl;
+		_type_log.flush();
+		
+		_size_log.seekp(0);
+		_size_log << "NSRC, NSEQ, NTIME = "
+				<< _nsrc << ", " 
+				<< _nseq_per_buf << ", "
+				<< _slot_ntime << endl;
+		_size_log.flush();
+	}
+	inline void close_logs() {
+		// Close the log files
+		_type_log.close();
+		_size_log.close();
+		_chan_log.close();
+		_stat_log.close();
+		
+		// Cleanup the process directory
+		int status = system(("rm -rf /dev/shm/bifrost/"+std::to_string(this->_pid)).c_str());
+		if( status != 0 ) {
+			// TODO: What to do here?
+			throw std::runtime_error("Cannot remove capture status directory");
```

Comment:
I'm not sure exactly what to do here either, but probably best to avoid throwing because this is essentially a destructor.